packages/contracts/src/contracts/current/test/TestLibMem/TestLibMem.sol

```diff
@@ -0,0 +1,231 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.24;
+
+import "../../utils/LibMem/LibMem.sol";
+import "../../utils/LibBytes/LibBytes.sol";
+
+contract TestLibMem is
```

Comment:
This implements the test logic in Solidity, which is a break from how we usually test this. Instead we could write a contract that exposes a function `TestMemCopy(bytes memory, uint256 from, uint256 to, uint256 length) returns (bytes memory)`.